### PR TITLE
Reward tests

### PR DESCRIPTION
**Pull request by ibrahim1023**
- Add tradable property to tokens
- Update maxToMint 
- Write tests for depositing and withdrawing non-tradable tokens

### CODE REVIEW

File: contracts/modules/FantomMintBalanceGuard.sol [hunk 1]
```diff
@@ -155,23 +197,28 @@ contract FantomMintBalanceGuard is FantomMintErrorCodes, IFantomMintBalanceGuard
...
+            collateralRatioDecimalsCorrection
+        );
 
         // if the account is under-collateralized already,
         // no tokens can be added
         if (cCollateralValue < minCollateralValue) {
             return 0;
         }
 
+        // what's the largest possible debt value allowed?
```
Comment (quan8) [inline contracts/modules/FantomMintBalanceGuard.sol:210]: ok

File: contracts/utility/FantomDeFiTokenStorage.sol [hunk 2]
```diff
@@ -109,52 +123,88 @@ contract FantomDeFiTokenStorage is Initializable, IFantomDeFiTokenStorage
...
+        bool requireTradable
+    ) internal view returns (uint256 value) {
         // loop all registered debt tokens
-        for (uint i = 0; i < tokens.length; i++) {
+        for (uint256 i = 0; i < tokens.length; i++) {
             // advance the result by the value of current token balance of this token.
             // Make sure to stay on safe size with the _sub deduction, we don't
             // want to drop balance to sub-zero amount, that would freak out the SafeMath.
             if (_token == tokens[i]) {
+                uint256 adjustedBalance = balance[_account][tokens[i]]
```
Comment (quan8) [inline contracts/utility/FantomDeFiTokenStorage.sol:182]: ok

File: contracts/utility/FantomDeFiTokenStorage.sol [hunk 3]
```diff
@@ -109,52 +123,88 @@ contract FantomDeFiTokenStorage is Initializable, IFantomDeFiTokenStorage
...
-                    ));
+                value = value.add(tokenValue(tokens[i], adjustedBalance));
 
                 // we consumed the adjustment and can reset it
                 _add = 0;
                 _sub = 0;
             } else {
-                // simply add the token balance converted to value as-is
-                value = value.add(tokenValue(tokens[i], balance[_account][tokens[i]]));
+                if (requireTradable) {
```
Comment (quan8) [inline contracts/utility/FantomDeFiTokenStorage.sol]: if (!requireTradable || addressProvider.getTokenRegistry().canTrade(tokens[i]))

Comment (ibrahim1023) [reply contracts/utility/FantomDeFiTokenStorage.sol]: I've tried this way. Doesn't let the test pass!

**Pull request merged**